internal/service/quicksight/data_set.go

```diff
@@ -0,0 +1,2800 @@
+package quicksight
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/quicksight"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceDataSet() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceAwsQuickSightDataSetCreate,
+		ReadWithoutTimeout:   resourceAwsQuickSightDataSetRead,
+		UpdateWithoutTimeout: resourceAwsQuickSightDataSetUpdate,
+		DeleteWithoutTimeout: resourceAwsQuickSightDataSetDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"aws_account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidAccountID,
+			},
+
+			"output_columns": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"output_column": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"description": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										Computed:     true,
+										ValidateFunc: validation.StringLenBetween(0, 500),
+									},
+									"name": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										Computed:     true,
+										ValidateFunc: validation.StringLenBetween(1, 128),
+									},
+									"type": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										Computed:     true,
+										ValidateFunc: validation.StringInSlice(quicksight.ColumnDataType_Values(), false),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
```

Comment:
Added to commit: "review-4"

- Removed `output_columns` from schema
- Created `waitDataSetCreated` function in `wait.go`
- Added check in create function using the logic in `waitDataSetCreated`

Reply:
Quick note: our preferred naming for waiter functions here would be just `DataSetCreated` and `DataSetUpdated` without the `wait` prefix ([see docs](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/retries-and-waiters.md#resource-lifecycle-waiters)). I noticed that we already have `waitCreated` defined for `quicksight/data_source` -- so I understand the confusion!

Regarding how to set up `DataSetUpdated`: do we know whether `output_columns` continues to output the same value while a Data Set is updating? Ideally, attempting to read `output_columns` might return a null value or an error during an update -- which we could use as a status check.

Otherwise, I'm not entirely sure how we can build a robust waiter without embedding some sort of "status" column as you're suggesting -- but we'd really prefer not to have Terraform act directly on operators' data in any way.